### DISCUSSION TITLE
For anyone else who can't install because of tinycuda

### DISCUSSION THREAD

**Discussion by kronkinatorix** [threestudio-project/threestudio#352] [General]
I found a copy of a wheel file for tinycudann after like 8 hours of searching for a way to install the damn thing.

It's for python3.10 but installing it in my venv allowed threestudio to install without hiccups. 

Not sure how it'll WORK after all this, but it's something. 

https://huggingface.co/camenduru/CoDeF/resolve/main/tinycudann-1.7-cp310-cp310-linux_x86_64.whl

**Comment by quarterturn** [comment]
Also, you can specify multiple CUDA capabilities at build time, like so:
```
TCNN_CUDA_ARCHITECTURES=60,70,72,75,80,86 pip install --force-reinstall --no-cache-dir git+https://github.com/NVlabs/tiny-cuda-nn/#subdirectory=bindings/torch
```
and build it from the source.

**Comment by sean1832** [comment]
I simply got it working by adding following to the `~/.bashrc`, after installing `CUDA 12.1` and pytorch.
```
export PATH="/usr/local/cuda/bin:$PATH" 
export LD_LIBRARY_PATH="/usr/local/cuda/lib64:$LD_LIBRARY_PATH" 
export LIBRARY_PATH="/usr/local/cuda/lib64/stubs:$LIBRARY_PATH"
```
make sure your `nvcc` is `12.1`
```
nvcc --version
```
